x-pack/solutions/observability/plugins/apm/server/routes/diagnostics/service_map/get_exit_spans_from_node.ts

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { termQuery, rangeQuery, existsQuery, termsQuery } from '@kbn/observability-plugin/server';
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import {
+  SPAN_DESTINATION_SERVICE_RESOURCE,
+  SPAN_ID,
+  PARENT_ID,
+  SERVICE_NAME,
+} from '@kbn/apm-types';
+import type { APMEventClient } from '@kbn/apm-data-access-plugin/server';
+
+/**
+ * Fetch exit spans for a given service, grouped by destination resource, with a sample doc for each.
+ * @param start Start time (epoch ms)
+ * @param end End time (epoch ms)
+ * @param serviceName Service name to filter on
+ */
+export async function getExitSpansFromSourceNode({
+  apmEventClient,
+  start,
+  end,
+  sourceNode,
+  destinationNode,
+}: {
+  apmEventClient: APMEventClient;
+  start: number;
+  end: number;
+  sourceNode: { field: string; value: string };
+  destinationNode: { field: string; value: string };
+}) {
+  const response = await apmEventClient.search('diagnostics_get_exit_spans_from_node', {
```

Comment:
I think we need to break this query in 2 because when the relationship is between services, the destination doc will be in the transaction docs, whereas the `destination` expected to be present on the same span doc.

We probably need to do something similar to what the service map does  https://github.com/elastic/kibana/blob/main/x-pack/solutions/observability/plugins/apm/server/routes/service_map/fetch_exit_span_samples.ts#L43